test/bun.js/bun-server.test.ts

```diff
@@ -26,4 +26,28 @@ describe("Server", () => {
     expect(await response.text()).toBe("Hello");
     server.stop();
   });
+
+  test('abort signal on server', async ()=> {
+    {
+      let signalOnServer = false;
+      const server = Bun.serve({
+        async fetch(req) {
+          req.signal.addEventListener("abort", () => {
+            signalOnServer = true;
+          });
+          await Bun.sleep(3000);
```

Comment:
can we make these timeouts shorter so the test doesn't take as long, but still assert sequential order?

Reply:
after that i think its good to merge